doc/ifa/SONiC_InbandFlowAnalyzer_FeatureSpec_Ver1.1.md

```diff
@@ -0,0 +1,478 @@
+
+1.4 Egress device is responsible for terminating the IFA flow by summarizing the telemetry data of the entire path and sending it to collector.
+
+2.0.0 IFA session needs configuration as mentioned below.
+
+2.0.0.1 Device type to specify whether INT header should be inserted or deleted.
+
+2.0.0.2 ACL configuration to identify a flow and sample packets from that flow to insert IFA headers.
+
+2.0.0.3 Collector configuration on egress device to forward telemetry data.
+
+2.0.1 The IFA feature will be a part of the TAM container along with other TAM features.
+
+3.0 UI commands available to configure IFA configuration.
+
+3.1 UI commands available to show IFA configuration, status and statistics.
+
+3.2 UI commands available to clear IFA configuration.
+
+4.0 The maximum number of IFA flows are platform dependent.
+
+4.1 Only one collector can be configured in a device.
+
+4.2 The IFA feature is supported on physical and LAG ports.
+
+4.3 The maximum packet length supported by IFA is platform dependent.
+
+4.4 The IFA feature does not support dynamic port breakout feature, however it can be manually configured on broken-out ports. The IFA feature does not automatically adjust configuration in response to dynamic port breakout events.
+
+### 1.1.2 Configuration and Management Requirements
+
+The Inband flow analyzer feature will use the Python click framework for CLI. Once the management framework for CLI is available, CLI will be changed to align to the same.
+
+### 1.1.3 Scalability Requirements
+
+IFA is scaled based on availability of resources in hardware such as ACLs, TAM object ids etc.
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+
+The Inband flow analyzer feature is newly developed. It is a static configuration feature, adapting the CONFIG_DB directly on top of the associated OrchAgent. There is no manager component involved.
+
+### 1.2.2 Container
+No new containers are added for Inband Flow Analyzer (IFA) feature.
+
+### 1.2.3 SAI Overview
+
+The SAI TAM spec specifies the TAM APIs to be used to configure the IFA functionality. Please refer to SAI-Proposal-TAM2.0-v2.0.docx in [https://github.com/opencomputeproject/SAI/tree/master/doc/TAM](https://github.com/opencomputeproject/SAI/tree/master/doc/TAM) for more details.
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+IFA is used to proactively monitor the network for faults and performance bottlenecks, it aids in detecting data plane faults and isolating them to a given device or location and assists in planning network capacity and projecting future load.
+
+## 2.2 Functional Description
+
+The IFA feature allows user to configure a device as ingress, intermediate and egress nodes through PACKET\_ACTION field in ACL rule. PACKET\_ACTION field values INT\_INSERT, INT\_UPDATE and INT\_DELETE specifies ingress, intermediate and egress devices respectively. Each device type performs its own unique operations as mentioned below.
```

Comment:
Can the spec expose the ASIC capabilities to the user ?

Reply:
Hi,
There is a PR submitted by Vissu with regard to exposing ASIC's capabilities, opencomputeproject/SAI#989. This issue can get addressed as a part of the same PR.